src/backend/distributed/commands/function.c

```diff
@@ -174,50 +193,114 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	MarkObjectDistributed(&functionAddress);
 
-	if (distributionArgumentName == NULL)
+	if (distributionArgumentName != NULL)
 	{
-		/* cannot provide colocate_with without distribution_arg_name */
-		if (pg_strncasecmp(colocateWithTableName, "default", NAMEDATALEN) != 0)
-		{
-			char *functionName = get_func_name(funcOid);
+		DistributeFunctionWithDistributionArgument(funcOid, distributionArgumentName,
+												   distributionArgumentOid,
+												   colocateWithTableName,
+												   &functionAddress);
+	}
+	else if (!colocatedWithReferenceTable)
```

Comment:
I am just writing this as a thought here, no need to update the PR.

I find this kind of assumptions a bit unsafe such as doing the `isDistributedTable` check with a `!colocatedWithReferenceTable`. Because the assumption `a table that is not a reference table will be distributed table` might fail at some point, however if we always do the right check, such as checking `isDistributedTable` or similar checks that will be much safer.